doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,150 @@
+Let's have a look at the documentation for std::StrictTotallyOrdered. Can you guess what it describes?
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, in order to model std::StrictTotallyOrdered we need to make our alphabet
+[equality comparable](\ref std::EqualityComparable) by implementing the (in-)equality operators (as free functions).
+\snippet alphabet_custom_steps.cpp equality
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+\snippet alphabet_custom_steps.cpp equality_comparable
+
+\assignment{Excercise}
+Implement the four compare operators *as free functions* and verify that your type models std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp comparison
+\endsolution
+
+## Semialphabet
+
+Let's move on to the more interesting concepts. seqan3::Semialphabet requires the *rank* interface
+that we have introduced in the [alphabet tutorial](\ref tutorial_alphabets). If we define the member functions
+`to_rank`, `assign_rank` and `alphabet_size` for our type, SeqAn automatically exposes them to the respective global
+functions `seqan3::to_rank`, `seqan3::assign_rank_to` and `seqan3::alphabet_size`. 
+Note that `assign_rank` is only required for seqan3::WritableSemialphabet.
+\snippet alphabet_custom_steps.cpp writable_semialphabet
+
+## Alphabet
+
+Now that you have a feeling for concepts, have a look at seqan3::Alphabet and make your type 
+also model the seqan3::Alphabet concept. If you keep the suggested namings, SeqAn will automatically expose
+the members to the respective global functions that model the concept.
+
+\assignment{Excercise}
+Please implement the member function `to_char(void)` which returns the char 'S' or 'W'
+depending on the rank value.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp alphabet
+\endsolution
+
+\assignment{Excercise}
+Now model the full seqan3::WritableAlphabet concept.
+
+1. Implement the member function `dna2 & assign_char(char const)`.
+2. Implement the static member function `bool char_is_valid(char const)`.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp writable_alphabet
+\endsolution
+
+At this point the seqan3::WritableAlphabet concept should be modelled successfully:
+\snippet alphabet_custom_steps.cpp writable_alphabet_concept
+
+## Shortcut: Alphabet base template
+
+In reality, you do not need to define all the functions you have learned in this exercise manually.
+Instead, you can inherit your type from seqan3::alphabet_base and just define the char-rank conversion
+in both directions. Read the documentation of seqan3::alphabet_base for details and examples.
+
```

Comment:
I would add the example of how dna2 would look if we would inherit alphabet_base.